lib/statesman/adapters/active_record_queries.rb

```diff
@@ -47,8 +47,20 @@ def included(base)
             query_builder.most_recent_transition_join
```

Comment:
Slight outside of the diff but can we ensure that this method hasn't changed, from memory this is covered by tests though

Reply:
@thom-oman Do you mean it should return SQL string?

Reply:
Lets decide whether you want to use SQL string everywhere or arel in some cases.

Reply:
My concern is that it'll be a breaking change for those expecting this method to return a string. One approach would be to have an optional parameter that returns arel, but defaults to string. Not sold on that idea though haha.

Reply:
Another solution is to add breaking change to the CHANGELOG. I think in 95% of the cases it will still work with arel, as those objects are still usable in `where`, `having` and `joins`. I don't think anyone concatenated string of a `JOIN`.

Reply:
Yeah I don't think it will be a concern for many, I can't think of many uses but I'm just thinking of [Hyrums laws](https://www.hyrumslaw.com/), not everyone checks the change log before bumping gems.

Also I just checked and this would break something in the GoCardless API, only one instance but you can assume it would break for others. Hence we should avoid making this a breaking change, don't think this justifies a major bump either.

Reply:
ARel considered as private interface for activerecord, so I don't think it should be publicly accessible by the Hyrums law. Going to make it to return SQL with a `.to_sql` call, so it will not to brake anyone code. Rebased with push force.